src/cdk/a11y/focus-trap.ts

```diff
@@ -45,9 +46,9 @@ export class FocusTrap {
 
   constructor(
     private _element: HTMLElement,
-    private _platform: Platform,
     private _checker: InteractivityChecker,
     private _ngZone: NgZone,
+    private _document: any,
```

Comment:
I'm not really happy with the propagation of `any` from this pattern. I just tried this locally and it works:

```ts
private _document: Document;

class Whatever {
  constructor(@Optional() @Inject(DOCUMENT) doc: any) {
    this._document = doc;
  }
}
```

It's a little more boilerplate but I think it's worth it. Start doing this instead?

Reply:
Makes sense. I'll switch these changes over.